### PR TITLE
Implement config and refactor logging middleware

### PR DESCRIPTION
**Pull request by jarri-abidi**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-	return Config{
-		ServerAddress:           "0.0.0.0:8085",
-		ServerWriteTimeout:      15 * time.Second,
-		ServerReadTimeout:       15 * time.Second,
-		ServerIdleTimeout:       60 * time.Second,
-		GracefulShutdownTimeout: 30 * time.Second,
-	}, nil
+// Load returns configuration from file or environment variables.
+// Values from file take precedence over environment variables.
+func Load(path string) (*Config, error) {
```
Comment (markhaur) [inline config/config.go:23]: it's good you are reading config variables from config file. Don't you think we should also add support to read from environment variables too?

**Review by markhaur — APPROVED** [review]
lgtm

File: config/config.go (hunk 1 above)
Comment (jarri-abidi) [reply config/config.go:23]: image
already taken care of

Comment (markhaur) [reply config/config.go:23]: (y)

**Pull request merged**